### DISCUSSION TITLE
Scheduling tasks in BB

### DISCUSSION THREAD

**Discussion by Hajelarohit** [Budibase/budibase#6094] [Help]
Hi BB Team
I am trying to create automated tasks based on a schedule.

For example, there is a meeting that needs to happen every month. The app should be able to create a new task every month for the next month for the user, when the meeting is due, the user changes the status to Completed and updates the meeting minutes and wait for the next month's task instead of manually creating it. If the meeting is supposed to occur on a quarterly basis, then the task should be created once every three months based on a already described schedule.

I am unable to do with with Automations or CRON job.

**Comment by melohagan** [comment]
Hi @Hajelarohit 

I think in this scenario you would have two automations roughly as follows:

#### Monthly meetings
1. Cron job trigger -> every month
2. Query Rows -> Filter on meetingType === "monthly" && status === "completed" && nextMeetingScheduled === false
3. Create Row -> (use <a href="https://docs.budibase.com/docs/automation-actions#looping">looping</a> for each row from the query) create your new tasks
4. Update Row -> Update the rows from the query again using looping, set `nextMeetingScheduled = true` 

#### Quarterly meetings
1. Cron job trigger -> every quarter
2. Query Rows -> Filter on meetingType === "quarter" && status === "completed" && nextMeetingScheduled === false
3. Create Row -> (use <a href="https://docs.budibase.com/docs/automation-actions#looping">looping</a> for each row from the query) create your new tasks
4. Update Row -> Update the rows from the query again using looping, set `nextMeetingScheduled = true` 

---

Please note that the query filter is currently broken: https://github.com/Budibase/budibase/issues/6090
We're looking to get that fixed as soon as possible.
